File: Labs/Week8/csvExample.py
#!/usr/bin/python
# csvExample.py - simple example, parsing a CSV file
#
# Kurt Schmidt
#
# Oct '07
#
# DEMONSTRATES - csv, csv.reader, sys.argv
#
# NOTES: reads stdin, or a (single) file from the command line
#

import csv, sys

if len( sys.argv ) < 2 : # read stdin
	fh = sys.stdin
else :
	try :
		fh = file( sys.argv[1] )
	except IOError, e :
		print "%s:  Could not open %s:  %s\n" % \
			( sys.argv[0], sys.argv[1], e[1] )
		sys.exit( 1 )

input = csv.reader( fh )

for l in input :
	for field in l :
		print field
	print

File: Labs/Week8/csvExample.pl
#!/usr/bin/perl
# csvExample.pl - simple example, parsing a CSV file
#
# Kurt Schmidt
#
# Mar '06
#
# DEMONSTRATES - module Text, module Text::CSV, <>, $_, array
#


use strict;
use warnings;

use Text::CSV_XS;

my $csv = Text::CSV_XS->new();              # create a new object

while( <> )
{
	my $status = $csv->parse( $_ );         # parse a CSV string into fields
	my @columns = $csv->fields();            # get the parsed fields

	foreach( @columns )
	{
		print $_."\n";
	}
	print "\n";
}

